### PR TITLE
Cleanup of `invalid input syntax for uuid xxx` errors

### LINKED ISSUE #3096: Cleanup of `invalid input syntax for uuid xxx` errors
After enforcing Visualization.id type we have now some errors populating the error log due to the old logic of "legacy endpoint checks first if id is a valid visualization, else if is a valid tablename".

Apply a regexp or similar to not even touch the Visualization model if hasn't got UUID format

### PR DESCRIPTION
**Pull request by Kartones**
Closes #3096

### CODE REVIEW

**Comment by Kartones** [comment]
@rafatower CR please

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2971/))

File: app/controllers/api/json/imports_controller.rb [hunk 1]
```diff
@@ -27,6 +27,11 @@ def index
   end
 
   def show
+    # If not valid uuid format out (avoid pollution from clients doing experiments)
+    if (params[:id] =~ /^[a-f0-9]{8}\-[a-f0-9]{4}\-[a-f0-9]{4}\-[a-f0-9]{4}\-[a-f0-9]{12}$/i) != 0
```
Comment (rafatower) [inline app/controllers/api/json/imports_controller.rb]: we have the gem `uuidtools` so we can take advantage of that:
```Ruby
irb(main):001:0> require 'uuidtools'
=> true

irb(main):010:0> UUIDTools::UUID.parse("hola")
ArgumentError: Invalid UUID format.
	from /var/lib/gems/1.9.1/gems/uuidtools-2.1.5/lib/uuidtools.rb:160:in `parse'
	from (irb):10
	from /usr/bin/irb:12:in `<main>'
irb(main):011:0> UUIDTools::UUID.parse(SecureRandom)
SecureRandom

irb(main):011:0> UUIDTools::UUID.parse(SecureRandom.uuid)
=> #<UUID:0xe6e1c8 UUID:edc5c12c-8215-4081-8ef1-b0e62ea1527a>

irb(main):013:0> UUIDTools::UUID_REGEXP
=> /^([0-9a-f]{8})-([0-9a-f]{4})-([0-9a-f]{4})-([0-9a-f]{2})([0-9a-f]{2})-([0-9a-f]{12})$/
```

my 2 cents

**Comment by Kartones** [comment]
ohh, I went the craftmanship way and checked the RFC instead of checking the gem :trollface: 
I'll change that, thanks!

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2981/))

**Pull request merged**